src/Microsoft.AspNet.OData/Batch/UnbufferedODataBatchHandler.cs

```diff
@@ -47,54 +47,56 @@ public override async Task<HttpResponseMessage> ProcessBatchAsync(HttpRequestMes
-            request.RegisterForDispose(reader);
+            using (ODataMessageReader reader = await request.Content.GetODataMessageReaderAsync(requestContainer, cancellationToken))
+            {
+                request.RegisterForDispose(reader);
 
-            ODataBatchReader batchReader = await reader.CreateODataBatchReaderAsync();
-            List<ODataBatchResponseItem> responses = new List<ODataBatchResponseItem>();
-            Guid batchId = Guid.NewGuid();
+                ODataBatchReader batchReader = await reader.CreateODataBatchReaderAsync();
+                Guid batchId = Guid.NewGuid();
 
-            HttpConfiguration configuration = request.GetConfiguration();
-            bool enableContinueOnErrorHeader = (configuration != null)
-                ? configuration.HasEnabledContinueOnErrorHeader()
-                : false;
+                HttpConfiguration configuration = request.GetConfiguration();
+                bool enableContinueOnErrorHeader = (configuration != null)
+                    ? configuration.HasEnabledContinueOnErrorHeader()
+                    : false;
 
-            SetContinueOnError(new WebApiRequestHeaders(request.Headers), enableContinueOnErrorHeader);
+                SetContinueOnError(new WebApiRequestHeaders(request.Headers), enableContinueOnErrorHeader);
 
-            try
-            {
-                while (await batchReader.ReadAsync())
+                try
                 {
-                    ODataBatchResponseItem responseItem = null;
-                    if (batchReader.State == ODataBatchReaderState.ChangesetStart)
-                    {
-                        responseItem = await ExecuteChangeSetAsync(batchReader, batchId, request, cancellationToken);
-                    }
-                    else if (batchReader.State == ODataBatchReaderState.Operation)
-                    {
-                        responseItem = await ExecuteOperationAsync(batchReader, batchId, request, cancellationToken);
-                    }
-                    if (responseItem != null)
+                    while (await batchReader.ReadAsync())
                     {
-                        responses.Add(responseItem);
-                        if (responseItem.IsResponseSuccessful() == false && ContinueOnError == false)
+                        ODataBatchResponseItem responseItem = null;
+                        if (batchReader.State == ODataBatchReaderState.ChangesetStart)
                         {
-                            break;
+                            responseItem = await ExecuteChangeSetAsync(batchReader, batchId, request, cancellationToken);
+                        }
+                        else if (batchReader.State == ODataBatchReaderState.Operation)
+                        {
+                            responseItem = await ExecuteOperationAsync(batchReader, batchId, request, cancellationToken);
```

Comment:
Not 100% sure why we don't apply `ConfigureAwait(false)` here. This can be fixed in a separate PR if necessary